hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/EvmTxProcessor.java

```diff
@@ -237,6 +251,54 @@ protected TransactionProcessingResult execute(
         }
     }
 
+    private Wei getTotalAllowanceCharged(
+            final long maxGasAllowance,
+            final Wei allowanceChargedForGasCosts,
+            final MutableAccount relayer) {
+        if (relayer == null) {
+            return allowanceChargedForGasCosts;
+        }
+        var totalAllowanceCharged = Wei.of(0).add(allowanceChargedForGasCosts);
+        final var remainingAllowanceAfterEvmEx =
+                (AllowanceWrapper)
+                        initialFrame.getContextVariable(FrameContextVariables.REMAINING_ALLOWANCE);
+        if (remainingAllowanceAfterEvmEx != null) {
+            final var initialAllowanceInEvm =
+                    maxGasAllowance - allowanceChargedForGasCosts.toLong();
+            final var spentAllowanceDuringExecution =
+                    initialAllowanceInEvm - remainingAllowanceAfterEvmEx.getValue();
+            if (spentAllowanceDuringExecution > 0) {
+                updater.getOrCreateSenderAccount(relayer.getAddress())
+                        .getMutable()
+                        .decrementBalance(Wei.of(spentAllowanceDuringExecution));
+                totalAllowanceCharged = totalAllowanceCharged.add(spentAllowanceDuringExecution);
+            }
+        }
+        return totalAllowanceCharged;
+    }
+
+    protected Map<String, Object> createContextVariables() {
+        final var globalDynamicProperties = (GlobalDynamicProperties) dynamicProperties;
+        final var knownRelayers = globalDynamicProperties.getKnownRelayers();
+        final var chargeFromAllowance =
+                lazyCreateChargingContext.isEthTxn()
+                        && (knownRelayers.isEmpty()
+                                || knownRelayers.contains(
+                                        lazyCreateChargingContext.relayer().getId().num()))
+                        && ((GlobalDynamicProperties) dynamicProperties)
+                                .isEvmLazyCreationGasChargeRemoved();
+
+        return Map.of(
+                FrameContextVariables.HEDERA_FUNCTIONALITY,
+                getFunctionType(),
+                FrameContextVariables.LAZY_CREATE_CHARGE_FROM_ALLOWANCE,
```

Comment:
We can remove this context variable and solely check if `REMAINING_ALLOWANCE != null`to determine whether we should charge from `allowance`/`gasLimit`.